classes/Project/ProjectFactory.js

```diff
@@ -181,6 +184,284 @@ export default class ProjectFactory {
+    const copiedGroup = await this.cloneGroup(project, creator, { 'Group Members': true })
+    return await new Project().create({ ...copiedProject, creator, group: copiedGroup._id })
+  }
+
+  static async cloneWithCustomizations(projectId, creator, modules) {
+    if (!projectId) {
+      throw {
+        status: 400,
+        message: "No project ID provided"
+      }
+    }
+
+    if (!modules || typeof modules !== 'object') {
+      throw {
+        status: 400,
+        message: "Modules must be an object"
+      }
+    }
+
+    const project = await ProjectFactory.loadAsUser(projectId, creator)
+    if (!project) {
+      throw {
+        status: 404,
+        message: "Project not found"
+      }
+    }
+
+    let copiedProject = this.copiedProjectConfig(project, database, creator, { 'Metadata': modules['Metadata'], 'Tools': modules['Tools'] })
+    let result = {}
+
+    if (modules['Layers'] && Array.isArray(modules['Layers']) && modules['Layers'].length > 0) {
+      for (const layer of project.layers) {
+        for (const newlayer of modules['Layers']) {
+          if (newlayer.hasOwnProperty(layer.id)) {
+            result[layer.id] = newlayer[layer.id].withAnnotations
+            break
+          }
+          else {
+            result[layer.id] = undefined
+          }
+        }
+
+        if (result[layer.id] === undefined) {
+          continue
+        }
+
+        const newLayer = {
+          id: `${process.env.SERVERURL}project/${copiedProject._id}/layer/${database.reserveId()}`,
+          label: layer.label,
+          pages: []
+        }
+
+        let newPages = []
+
+        if(result[layer.id]) {
+          newPages = await this.clonePages(layer, copiedProject, database, true)
+          newLayer.pages.push(...newPages)
+          copiedProject.layers.push(newLayer)
+        }
+
+        if(!result[layer.id]) {
+          newPages = await this.clonePages(layer, copiedProject, database, false)
+          newLayer.pages.push(...newPages)
+          copiedProject.layers.push(newLayer)
+        }  
+      }
+    }
+    else {
+      const newLayer = {
+        id: `${process.env.SERVERURL}project/${copiedProject._id}/layer/${database.reserveId()}`,
+        label: project.layers[0].label,
+        pages: []
+      }
+      const newPages = await this.clonePages(project.layers[0], copiedProject, database, false)
+      newLayer.pages.push(...newPages)
+      copiedProject.layers.push(newLayer)
+    }
+
+    modules['Group Members'].push(creator)
+    const copiedGroup = await this.cloneGroup(project, creator, { 'Group Members': modules['Group Members'] })
+    if( modules['Hotkeys'] ) {
+      await this.cloneHotkeys(project._id, copiedProject._id)
+    }
+    copiedProject._lastModified = copiedProject.layers[0]?.pages[0]?.id.split('/').pop()
+    return await new Project().create({ ...copiedProject, creator, group: copiedGroup._id })
```

Comment:
This is the right idea, but this isn't a "create" action. The existing project keys can be brought over and we'll know it is valid. We can still run the validation from create() if we want, but once we have the copied project, database.save() should be enough.